packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react'
+import PropTypes from 'react-proptypes'
+
+import { withMutations } from 'cozy-client'
+
+import AccountCreator from './AccountCreator'
+import AccountEditor from './AccountEditor'
+import TriggerSuccessMessage from './TriggerSuccessMessage'
+import { triggersMutations } from '../connections/triggers'
+import {
+  buildKonnectorCron,
+  buildKonnectorTriggerAttributes
+} from '../helpers/triggers'
+
+const IDLE = 'IDLE'
+const RUNNING = 'RUNNING'
+const LOGGED = 'LOGGED'
+const SUCCESS = 'SUCCESS'
+
+/**
+ * Deals with konnector configuration, i.e encapsulate account creation or
+ * edition
+ * @type {Component}
+ */
+export class TriggerManager extends Component {
+  state = {
+    status: IDLE
+  }
+
+  /**
+   * Creation/update start handler
+   * Set the status to RUNNING as soon as an account is being created
+   * or updated.
+   */
+  handleAccountMutation = () =>
+    this.setState({
+      status: RUNNING
+    })
+
+  /**
+   * Account creation success handler
+   * @param  {Object}  account Created io.cozy.accounts document
+   * @return {Object}          io.cozy.jobs document, runned with account data
+   */
+  handleAccountCreationSuccess = async account => {
+    const { createTrigger, konnector } = this.props
+
+    const trigger = await createTrigger(
+      buildKonnectorTriggerAttributes({
+        konnector,
+        account,
+        cron: buildKonnectorCron(konnector)
+      })
+    )
+
+    return await this.launch(trigger)
+  }
+
+  /**
+   * Account update success handler
+   * @param  {Object}  account Updated io.cozy.accounts document
+   * @return {Object}          io.cozy.jobs document, runned with account data
+   */
+  handleAccountUpdateSuccess = async account => {
+    this.setState({
+      updatedAccount: account
+    })
+
+    const { trigger } = this.props
+    return await this.launch(trigger)
+  }
+
+  /**
+   * Launches a trigger
+   * @param  {Object}  trigger io.cozy.triggers document
+   * @return {Promise}         [description]
+   */
+  launch = async trigger => {
+    const { launchTrigger, onLoginSuccess, waitForLoginSuccess } = this.props
+
+    const job = await waitForLoginSuccess(await launchTrigger(trigger))
+
+    if (['queued', 'running'].includes(job.state)) {
+      onLoginSuccess(trigger)
+      this.setState({
+        LOGGED
+      })
+    }
+
+    this.setState({
+      status: SUCCESS
+    })
```

Comment:
At the end of the function the status will always be success, I am not sure this is wanted, is there an early return missing ?

Reply:
The next step is to deal with errors. But you're right I need to think about the wanted status at the end of the process.